tests/test_jump.py

```diff
@@ -59,12 +47,192 @@ def test_find_simple_ellipse():
+    cube[1, 3, 2] = DQFLAGS['SATURATED']
+    cube[1, 2, 2] = DQFLAGS['JUMP_DET']
+    sat_circles = find_ellipses(cube[grp, :, :], DQFLAGS['SATURATED'], 1)
+    new_cube = extend_saturation(cube, grp, sat_circles, DQFLAGS['SATURATED'], DQFLAGS['JUMP_DET'],
+                                 min_sat_radius_extend, expansion=1.1)
+
+    assert new_cube[grp, 2, 2] == DQFLAGS['SATURATED']
+    assert new_cube[grp, 4, 4] == DQFLAGS['SATURATED']
+    assert new_cube[grp, 4, 5] == 0
+
+
+def test_flag_large_events_nosnowball():
+    cube = np.zeros(shape=(1, 5, 7, 7), dtype=np.uint8)
+    grp = 1
+    min_sat_radius_extend = 1
+    # cross of saturation with no jump
+    cube[0, 1, 3, 3] = DQFLAGS['SATURATED']
+    cube[0, 1, 2, 3] = DQFLAGS['SATURATED']
+    cube[0, 1, 3, 4] = DQFLAGS['SATURATED']
+    cube[0, 1, 4, 3] = DQFLAGS['SATURATED']
+    cube[0, 1, 3, 2] = DQFLAGS['SATURATED']
+    # cross of saturation surrounding by jump -> snowball but sat core is not new
+    # should have no snowball trigger
+    cube[0, 2, 3, 3] = DQFLAGS['SATURATED']
+    cube[0, 2, 2, 3] = DQFLAGS['SATURATED']
+    cube[0, 2, 3, 4] = DQFLAGS['SATURATED']
+    cube[0, 2, 4, 3] = DQFLAGS['SATURATED']
+    cube[0, 2, 3, 2] = DQFLAGS['SATURATED']
+    cube[0, 2, 1, 1:6] = DQFLAGS['JUMP_DET']
+    cube[0, 2, 5, 1:6] = DQFLAGS['JUMP_DET']
+    cube[0, 2, 1:6, 1] = DQFLAGS['JUMP_DET']
+    cube[0, 2, 1:6, 5] = DQFLAGS['JUMP_DET']
+    fits.writeto('input_flag_large_events.fits', cube, overwrite=True)
+    flag_large_events(cube, DQFLAGS['JUMP_DET'], DQFLAGS['SATURATED'], min_sat_area=1,
+                      min_jump_area=6,
+                      expand_factor=1.9, use_ellipses=False,
+                      sat_required_snowball=True, min_sat_radius_extend=1, sat_expand=1.1)
+    fits.writeto("output_flag_large_events.fits", cube, overwrite=True)
+    assert cube[0, 1, 2, 2] == 0
+    assert cube[0, 1, 3, 5] == 0
+
+
+def test_flag_large_events_withsnowball():
+    cube = np.zeros(shape=(1, 5, 7, 7), dtype=np.uint8)
+    grp = 1
+    min_sat_radius_extend = 1
+    # cross of saturation surrounding by jump -> snowball
+    cube[0, 2, 3, 3] = DQFLAGS['SATURATED']
+    cube[0, 2, 2, 3] = DQFLAGS['SATURATED']
+    cube[0, 2, 3, 4] = DQFLAGS['SATURATED']
+    cube[0, 2, 4, 3] = DQFLAGS['SATURATED']
+    cube[0, 2, 3, 2] = DQFLAGS['SATURATED']
+    cube[0, 2, 1, 1:6] = DQFLAGS['JUMP_DET']
+    cube[0, 2, 5, 1:6] = DQFLAGS['JUMP_DET']
+    cube[0, 2, 1:6, 1] = DQFLAGS['JUMP_DET']
+    cube[0, 2, 1:6, 5] = DQFLAGS['JUMP_DET']
+    fits.writeto('input_flag_large_events.fits', cube, overwrite=True)
+    flag_large_events(cube, DQFLAGS['JUMP_DET'], DQFLAGS['SATURATED'], min_sat_area=1,
+                      min_jump_area=6,
+                      expand_factor=1.9, use_ellipses=False, edge_size=0,
+                      sat_required_snowball=True, min_sat_radius_extend=.5, sat_expand=1.1)
+    fits.writeto("output_flag_large_events.fits", cube, overwrite=True)
```

Comment:
does this need to be saved?

Reply:
I removed all the fits.writeto except in the two tests that are skipped.